### ISSUE TITLE
Concatenation of features

### ISSUE THREAD

**Issue by JovanVeljanoski** [vaexio/vaex#538]
There are a number of cases in ML (e.g. fraud detection, click-through-rates) in which one may want to create a feature by simply concatenating existing features (or columns). If the features are of type `str` this is already possible. If we have a combination of numeric and string data, this is right now not possible out of the box, and not optimized. 

Is it useful to implement a DataFrame method that will do fast concatenation of features, regardless of their data type (for primitive types). 

In addition to this, consider this example:
```
import vaex
import numpy as np

x = np.arange(10)
y = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
df = vaex.from_arrays(x=x, y=y)
```

Now, doing an operation such as `df.y + df.x` results in an error, which is expected since we are mixing dtypes. 

However, doing `df.x + df.y` gives an answer:
![image](https://user-images.githubusercontent.com/18574951/71589663-f1477d80-2b25-11ea-9631-1c9671429280.png)
instead of raising an exception.

I guess we should either raise an exception or cast to `str` behind the scenes in this case. Whatever the decision, the behaviour should be the same as in the 1st example above.

Finally, doing a "manual" concatenation of numeric and str features (of the order of ~20 in total) like this:
```
df[concat_feature] = df['str_feat1'] + df['str_feat2'] + df['num_feat1'].astype('str') + df['num_feat1'].astype('str') + ...
```
works but any operation done on the resulting `df.concat_feature` column such as `unique` or `value_counts` are quite slow. 

I propose we do some work here to improve on this.

**Comment by maartenbreddels** [comment]
> However, doing `df.x + df.y` gives an answer:

Indeed, I think this is a bug, please open a PR with a test that should raise an exception, thanks!

On the other hand, we should not cast to string automatically, we should stay close to python as possible. It is not clear to me what the issue is? If something is slow, we should understand/fix that.
